lib/ret_web/controllers/api/v1/project_controller.ex

```diff
@@ -0,0 +1,112 @@
+defmodule RetWeb.Api.V1.ProjectController do
+  import Ecto.Query
+
+  use RetWeb, :controller
+
+  alias Ret.{Account, Project, Repo, Storage}
+
+  # Limit to 1 TPS
+  plug(RetWeb.Plugs.RateLimit when action in [:create])
+
+  def index(conn, %{} = _params) do
+    account = Guardian.Plug.current_resource(conn)
+    projects = get_projects(account)
+    render(conn, "index.json", projects: projects)
+  end
+
+  def show(conn, %{"id" => project_sid}) do
+    case project_sid |> get_project() do
+      %Project{} = project -> render(conn, "show.json", project: project)
+      _ -> conn |> send_resp(404, "not found")
+    end
+  end
+
+  def create(conn, %{"project" => params}) do
+    account = conn |> Guardian.Plug.current_resource()
+
+    {result, project} =
+      %Project{}
+      |> Project.changeset(account, params)
+      |> Repo.insert_or_update()
+
+    project = Repo.preload(project, [:project_owned_file, :thumbnail_owned_file])
+
+    case result do
+      :ok ->
+        conn |> render("show.json", project: project)
+
+      :error ->
+        conn |> send_resp(422, "invalid project")
+    end
+  end
+
+  def update(conn, %{"id" => project_sid, "project" => params}) do
+    account = conn |> Guardian.Plug.current_resource()
+
+    case project_sid |> get_project() do
+      %Project{} = project -> update(conn, params, project, account)
+      _ -> conn |> send_resp(404, "not found")
+    end
+  end
+
+  defp get_projects(account) do
```

Comment:
I'd move these db interacting function into `Ret.Projects.projects_for_account` and `Ret.Projects.project_by_sid`